### PR TITLE
Permit boards to boot in case of sensor failure

### PR DESCRIPTION
**Pull request by peterbarker**
... also, if someone sets their board type incorrectly, this will allow them to correct their mistake

### CODE REVIEW

File: libraries/AP_BoardConfig/AP_BoardConfig.h [hunk 1]
```diff
@@ -122,6 +125,7 @@ class AP_BoardConfig {
     } px4;
 
     static enum px4_board_type px4_configured_board;
+    static bool _in_sensor_config_error;
```
Comment (OXINARF) [inline libraries/AP_BoardConfig/AP_BoardConfig.h]: This can't be inside a PX4/VRBrain-only block.

**Comment by OXINARF** [comment]
> also, if someone sets their board type incorrectly, this will allow them to correct their mistake

We should already allow that. Isn't it working?

I'm not against this, I don't prefer one or the other, but, if it hasn't yet, this should probably go through @tridge since I think he implemented it this way on purpose.

File: libraries/AP_BoardConfig/AP_BoardConfig.h (hunk 1 above)
Comment (peterbarker) [inline libraries/AP_BoardConfig/AP_BoardConfig.h]: Sod.  So much for last minute "it would look neater if I put it here" changes.

Fixed.

**Comment by peterbarker** [comment]
@OXINARF we did allow that, but a workaround for really-slow-parameter-reading-from-MP broke it: https://github.com/ardupilot/ardupilot/blob/master/ArduCopter/GCS_Mavlink.cpp#L863

**Comment by rmackay9** [comment]
I'm happy with the Copter change if there's any question.  More generally I think we should simply allow the board to boot up normally and then not allow arming but this change is also ok.

**Comment by OXINARF** [comment]
@peterbarker Ah the PR title threw me off. Of course this is just fixing the failed boot case. Merged, thanks!

@rmackay9 I think that would involve more changes to libraries, but specially would not allow you to use a board with a dead IMU for example.

**Pull request merged**